draft-ietf-rats-corim.md

```diff
@@ -1066,6 +1072,328 @@ The following describes each member of the `concise-bom-tag` map.
+ multiple entries in its `measurement-values-map`; or a combination of
+ these approaches.
+
+If evidence from different sources has the same `environment-map` then the
+`measurement-values-map`s are merged.
+If both measurement-value-maps being merged contain the same key then the
+values associated with that key MUST be binary identical.
+
+> A CoRIM profile MUST describe:
+>
+> * How evidence is converted to a format suitable for appraisal
+
+Section {sec-dice-spdm} provides information on how evidence collected using
+DICE and SPDM protocols is added to the Accepted Claims Map.
+
+
+## Evidence appraisal
+
+In the Evidence Appraisal phase, a CoRIM Appraisal Context and an Evidence
+Appraisal Policy are used by the Verifier to appraise the received evidence.
+If the evidence is acceptable then the CoRIM can supplement it with a more
+concise description of the attestor.
+For example, a CoRIM provided by a firmware author might match against the hash
+of a piece of firmware and provide the version number of that firmware.
+This phase may be repeated multiple times.
+The outcome of the appraisal process is summarised in an Attestation Result.
+The Relying Party application uses the content of the Attestation Result to
+make its own policy decisions.
+
+This specification makes no assumptions on the specific shape of the
+Attestation Result, except for its optional ability to include Evidence from
+the attestor and Endorsed Values that the Verifier has been able to infer from
+Evidence and the Appraisal Context.
+
+> A CoRIM profile MUST describe:
+>
+> * How Reference Values are conveyed (e.g., using Reference Value triples, or CoSWID tags)
+> * How Reference Values are associated with the Target Environment
+> * How the Target Environment is identified in Evidence
+> * How measurements from Evidence are matched against Reference Values
+
+> A CoRIM profile MUST describe:
+>
+> * How Endorsed Values are conveyed (e.g., using Endorsed Value triples, or Device Identity triples)
+> * How Endorsed Values are added to the Accepted Claims Map
+
+The sections below describe the algorithm used to compare CoMID tags.
+
+### Comparing and processing CoMID tags
+
+There can be multiple passes of the CoMID evidence appraisal process.
+
+In each pass, each potential CoMID tag in the Appraisal Context is compared
+against the accepted claims set (which has been initialised with Evidence from
+the attestor(s) ).
+If the reference values in the CoMID tag match the Accepted Claims Set then
+Endorsements and other values from the tag are added to the Accepted Claims Set.
+
+If any tags matched in a pass through the CoMID appraisal process then another
+pass is performed.
+Tags which matched in an earlier pass are ignored in later passes of the same
+appraisal.
+
+A later pass may match Reference Values from the CoMID against Endorsements
+added by matched tags in an earlier pass, this allows CoRIM authors to create
+hierarchical descriptions of a system.
```

Comment:
Complex behaviour as the one described above should be encoded in a triple rather than trickling through the whole process.